### PR TITLE
KYLIN-4067 optimize the kylin cube page loading mechanism.

### PR DESCRIPTION
**Pull request by zhaojintaozhao**
KYLIN-4067 optimize the kylin cube page loading mechanism. Get all information of cubes only when adding a new cube.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by zhaojintaozhao** [comment]
> ## Pull Request Test Coverage Report for [Build 4691](https://coveralls.io/builds/24350345)
> * **0** of **0**   changed or added relevant lines in **0** files are covered.
> * **3** unchanged lines in **2** files lost coverage.
> * Overall coverage increased (+**0.001%**) to **28.271%**
> 
> Files with Coverage Reduction	New Missed Lines	%
> [core-cube/src/main/java/org/apache/kylin/cube/inmemcubing/MemDiskStore.java](https://coveralls.io/builds/24350345/source?filename=core-cube%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fkylin%2Fcube%2Finmemcubing%2FMemDiskStore.java#L553)	1	78.12%
> [core-job/src/main/java/org/apache/kylin/job/impl/threadpool/DefaultScheduler.java](https://coveralls.io/builds/24350345/source?filename=core-job%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fkylin%2Fjob%2Fimpl%2Fthreadpool%2FDefaultScheduler.java#L194)	2	80.23%
> Totals	[Coverage Status](https://coveralls.io/builds/24350345)
> Change from base [Build 4690](https://coveralls.io/builds/24339755):	0.001%
> Covered Lines:	23411
> Relevant Lines:

> ## Pull Request Test Coverage Report for [Build 4691](https://coveralls.io/builds/24350345)
> * **0** of **0**   changed or added relevant lines in **0** files are covered.
> * **3** unchanged lines in **2** files lost coverage.
> * Overall coverage increased (+**0.001%**) to **28.271%**
> 
> Files with Coverage Reduction	New Missed Lines	%
> [core-cube/src/main/java/org/apache/kylin/cube/inmemcubing/MemDiskStore.java](https://coveralls.io/builds/24350345/source?filename=core-cube%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fkylin%2Fcube%2Finmemcubing%2FMemDiskStore.java#L553)	1	78.12%
> [core-job/src/main/java/org/apache/kylin/job/impl/threadpool/DefaultScheduler.java](https://coveralls.io/builds/24350345/source?filename=core-job%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fkylin%2Fjob%2Fimpl%2Fthreadpool%2FDefaultScheduler.java#L194)	2	80.23%
> Totals	[Coverage Status](https://coveralls.io/builds/24350345)
> Change from base [Build 4690](https://coveralls.io/builds/24339755):	0.001%
> Covered Lines:	23411
> Relevant Lines:	82808
> ##### 💛 - [Coveralls](https://coveralls.io)

I just modify js files, why the  "Test Coverage Report for Build 4691" show  tow java files reduce coverage ?

**Comment by zhaojintaozhao** [comment]
> it's better to add a special API to check the duplicate cube name

Hi, Guosheng:
That's a good idea. I add a new API to check the duplicate cube name.  
The main value of this PR is to speed up when open  Kylin web page. If one project has more than 200 cubes,  it loads very slowly when opening kylin cube page， even may open failure.

**Comment by nichunen** [comment]
Hi @zhaojintaozhao 
I'm afraid the cube list won't be updated if one cube be created/updated/deleted in another Kylin instance by this PR

File: webapp/app/js/controllers/cubeSchema.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ KylinApp.controller('CubeSchemaCtrl', function ($scope, QueryService, UserServic
     ];
 
     $scope.curStep = $scope.wizardSteps[0];
+    $scope.allCubeNames = [];
+    $scope.cubeService = CubeService;
```
Comment (nichunen) [inline webapp/app/js/controllers/cubeSchema.js]: $scope.cubeService seems never used

Comment (zhaojintaozhao) [reply webapp/app/js/controllers/cubeSchema.js]: Hi， chunen：
Yes, "$scope.cubeService"  never be used. 
I will submit code to delete this line tonight.

Comment (zhaojintaozhao) [reply webapp/app/js/controllers/cubeSchema.js]: Hi,  chunen:
I have deleted this unused line: "$scope.cubeService = CubeService" and submit code.
Please verify this patch, Thank you.

**Comment by RupengWang** [comment]
LGTM

**Pull request merged**